src/Logging/Logging.File/FileLogService.cs

```diff
@@ -1,45 +1,104 @@
-﻿using static CodeMonkeys.Argument;
+﻿using CodeMonkeys.Logging.Batching;
 
-using System;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
 
 namespace CodeMonkeys.Logging.File
 {
-    internal sealed class FileLogService : ILogService
+    /// <summary>
+    /// <see cref="IScopedLogService"/> which writes to one or multiple <see cref="System.IO.File"/>('s).
+    /// </summary>
+    public sealed class FileLogService : BatchLogService<FileLogOptions>
     {
-        private readonly FileLogServiceProvider _provider;
-        private readonly string _context;
+        private int _index;
 
-        internal FileLogService(
-            FileLogServiceProvider provider, 
-            string context)
+        internal FileLogService(string context)
+            : base(context)
         {
-            _provider = provider;
-            _context = context;
         }
 
-        public bool IsEnabledFor(LogLevel logLevel) => _provider.IsEnabledFor(logLevel);
+        protected override async Task PublishMessageBatch(IEnumerable<LogMessage> messageBatch)
+        {
+            Directory.CreateDirectory(Options.Directory);
+
+            foreach (var message in messageBatch)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Refactor this method to add validation of parameter 'messageBatch' before using it.](https://app.codacy.com/gh/UltimateCodeMonkeys/CodeMonkeysMVVM/pullRequest?prid=5626772)